### ISSUE TITLE
Improve error message when an invalid Maven coordinate is specified for a publication

### ISSUE THREAD

**Issue by adammurdoch** [gradle/gradle#3834]
When I attempt to publish a Maven publication with an invalid artifactId, for example, I get an error message something like this:

```
Invalid publication 'main': artifactId is not a valid Maven identifier ([A-Za-z0-9_\-.]+).
```

However, the message doesn't tell me _what_ my artifactId is and so it is difficult to diagnose exactly what I did wrong here, particularly when the value is calculated by something from some other input I gave.

### Expected Behavior

Include more context in the error message, such as the actual value that Gradle was given, and even better, where the value came from. And maybe a better explanation of the rules I need to conform to, rather than a regexp.

### Current Behavior

The error message is missing some important information to help me diagnose the problem.

**Comment by PLNech** [comment]
Any progress on this? That error is especially hard to debug on multi-module projects as the exact `artifactId` is hard to pinpoint. (I'm actually wondering if other situations can't trigger this error, as all my modules' artifactIds do match `[A-Za-z0-9_\-.]+` :thinking:)

**Comment by etiennestuder** [comment]
+1. I'm running into the same issue.

**Comment by stauvel** [comment]
For my case, it was extra quotes in my groupId name. I found it by debugging org.gradle.api.publish.maven.internal.publisher.ValidatingMavenPublisher.MavenFieldValidator#validMavenIdentifier.

May it help someone else...

**Comment by bigdaz** [comment]
Fixed in #9348
